packages/cli/README.md

```diff
@@ -127,3 +127,615 @@ jobs:
+  const fileSizeAudit = {
+    slug: 'file-size-audit',
+    title: 'File size audit',
+  };
+  return {
+    slug: 'file-size',
+    title: 'File size plugin',
+    icon: 'javascript',
+    audits: [fileSizeAudit],
+    runner: runnerFunction(options),
+  };
+}
+
+// we use a closure to pass options to the runner function for better DX
+async function runnerFunction(options: Options): RunnerFunction {
+  return () => {
+    // implementation follows in the below section
+    return [] as AuditOutputs;
+  };
+}
+
+export default {
+  persist: {
+    outputDir: '.code-pushup',
+  },
+  plugins: [
+    await create({
+      directory: './src',
+    }),
+  ],
+  categories: [],
+};
+```
+
+</details>
+
+The first thing we should think about is how to get the raw data to calculate the metrics.
+In our case we need to get the file name and size of all files in a directory.
+
+The basic implementation looks like this:
+
+**get file-size data**
+
+```typescript
+// code-pushup.config.ts
+
+// ...
+
+// get raw file size data
+type FileSizeInfo = { file: string; size: number };
+async function getFileSizeData(options: Options): Promise<FileSizeInfo[]> {
+  const { directory } = options;
+
+  let results = [];
+  const files = await readdir(directory);
+
+  for (const file of files) {
+    const filePath = join(directory, file);
+    const stats = await stat(filePath);
+
+    // if file, get file size info
+    if (stats.isFile()) {
+      results.push({ filePath, size: stats.size });
+    }
+    // if directory, recurse
+    else if (stats.isDirectory()) {
+      results.push(...(await getFileSizeData({ directory: filePath })));
+    }
+  }
+
+  return results;
+}
+```
+
+The above code will recursively get the file size info for all files in the specified directory and it's subdirectories.
+
+Let's create a runner function that uses the above code to get the file size data and turns it into the shape
+of `AuditOutputs`:
+
+**file-size `RunnerFunction`**
+
+```typescript
+// code-pushup.config.ts
+import { pluralize } from '@code-pushup/utils';
+
+// ...
+
+async function runnerFunction(options: Options): Promise<AuditOutputs> {
+  const data = await getFileSizeData(options);
+  // We use a helper function of the utils package to pluralize the display value
+  const displayValue = `${data.length} ${data.length === 1 ? 'file' : pluralize('file')}`;
+
+  let fileSizeAuditOutput: AuditOutput = {
+    slug: fileSizeAudit.slug,
+    // We have always a score of 1. Proper implementation of scores will be covered in a next section.
+    score: 1,
+    value: data.length,
+    displayValue,
+  };
+
+  return [fileSizeAuditOutput];
+}
+```
+
+Now we can execute the CLI with `npx code-pushup collect --no-progress` and see a similar output as the following:
+
+**stdout of basic file-size plugin**
+
+```sh
+Code PushUp Report - @code-pushup/core@x.y.z
+
+File size plugin audits
+● File size audit                                                            2
```

Comment:
At this point I believe it would be `2 files` instead of just `2`, because you've created the `displayValue`:

```suggestion
● File size audit                                                            2 files
```